src/mlpack/methods/time_series_models/persistence_impl.hpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * @file methods/time_series_models/persistence_impl.hpp
+ * @author Rishabh Garg
+ *
+ * Implementation of the Persistence Model for time series data.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_TIME_SERIES_MODELS_PERSISTENCE_MODEL_IMPL_HPP
+#define MLPACK_METHODS_TIME_SERIES_MODELS_PERSISTENCE_MODEL_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "persistence.hpp"
+#include "util/lag.hpp"
+
+namespace mlpack {
+namespace ts /* Time Series methods. */ {
+
+PersistenceModel::PersistenceModel() { /* Nothing to do here. */ }
```

Comment:
So, the reason you are getting a duplicate symbol error is that we can only define templated and inline functions safely in header files.  Otherwise, if they are included from multiple different places, the compiler will create multiple versions of them.  I think my explanation above is not the best, but maybe [this SO post](https://stackoverflow.com/questions/8201944/multiple-definition-and-header-only-libraries) could be helpful?  You could continue down the rabbit hole too and read about [translation units](https://stackoverflow.com/questions/1106149/what-is-a-translation-unit-in-c), if you wanted.  (Be careful: the C++ rabbit hole never ends!  There is always more to learn, which is both exciting and discouraging... :smiley:)